### PR TITLE
Get tests running under py3

### PR DESCRIPTION
**Pull request by tipabu**
While we're at it, fix up some tests:

- All our configs should be unicode since they came out of `json`.

- `find_new_rows()` isn't a thing any more; fix some mocks.

- `/dev/sda` might actually exist and not be a directory, tripping an `ENOTDIR` error. Use `/srv/node/sda` instead.

IDK how to travis, but we should probably make `tox -e py3-master` gating.

### CODE REVIEW

**Comment by tipabu** [comment]
Rebased to resolve a couple conflicts and address some py3-breakage.

**Comment by timuralp** [comment]
@tipabu I added the travis stuff to run python3 tests without using tox for the time being. It's a little annoying with the base image using system packages. We only have one virtualenv and it's arguably nicer to pre-setup all the dependencies at container build time... Anyway, thanks for tackling this!

File: test/unit/test_container_crawler.py [hunk 1]
```diff
@@ -63,15 +65,17 @@ def setUp(self):
             'status_dir': mkdtemp(),
             'containers': [{'account': 'account',
                             'container': 'container'}]}
+        # conf files get loaded from json; simulate that
```
Comment (timuralp) [inline test/unit/test_container_crawler.py]: Does this actually help us?

File: test/unit/test_container_crawler.py [hunk 2]
```diff
@@ -245,8 +248,8 @@ def my_num_from_row(row):
             with self._patch_broker():
                 self.crawler.run_once()
 
-            handle_rows = filter(lambda x: x % 2 == node_id, range(rows))
-            verify_rows = filter(lambda x: x % 2 != node_id, range(rows))
+            handle_rows = list(range(node_id, rows, 2))
+            verify_rows = list(range(1 - node_id, rows, 2))
```
Comment (timuralp) [inline test/unit/test_container_crawler.py]: Hm I feel like this is less obvious to parse than the prior check against `node_id`

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[tox]
```
Comment (timuralp) [inline tox.ini]: We usually test against the docker container swift version. We could not base on that image entirely and do something with tox and pull the code directly. Is this change required for the python3 improvements? I observed all of the tests passing against the latest Swift in docker-swift.

File: test/unit/test_container_crawler.py (hunk 1 above)
Comment (tipabu) [reply test/unit/test_container_crawler.py]: Yup; without it you get a bunch of failures on py2:

```
======================================================================
ERROR: test_prune_status_handles_rmtree_errors (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/.tox/py27-minver/lib/python2.7/site-packages/mock/mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1107, in test_prune_status_handles_rmtree_errors
    {u'fo\u00f5': [u'c\u00f5nt1', u'c\u00f5nt2']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1035, in _prune_files_test_helper
    self.crawler.run_once()
  File "/home/tburke/code/container-crawler/container_crawler/crawler.py", line 554, in run_once
    self._submit_containers()
  File "/home/tburke/code/container-crawler/container_crawler/crawler.py", line 538, in _submit_containers
    self._prune_status_files()
  File "/home/tburke/code/container-crawler/container_crawler/crawler.py", line 421, in _prune_status_files
    e))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 17: ordinal not in range(128)

======================================================================
FAIL: test_does_not_prune_per_account (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1090, in test_does_not_prune_per_account
    {u'fo\u00f5': [u'c\u00f5nt1', u'c\u00f5nt2']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the second set but not the first:
u'fo\xf5'

======================================================================
FAIL: test_does_not_prune_shards (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1080, in test_does_not_prune_shards
    u'.shards_blah-bla\u00f9': [u'b\u00e0r']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the first set but not the second:
'.shards_blah-bla\xc3\xb9'
Items in the second set but not the first:
u'fo\xf5'
u'.shards_blah-bla\xf9'

======================================================================
FAIL: test_prune_status_extra_files (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1135, in test_prune_status_extra_files
    ['other-status.file'])
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the second set but not the first:
u'fo\xf5'

======================================================================
FAIL: test_prune_status_files (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1055, in test_prune_status_files
    {u'fo\u00f5': [u'c\u00f5nt1']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the second set but not the first:
u'fo\xf5'

======================================================================
FAIL: test_prune_status_files_directory (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1071, in test_prune_status_files_directory
    {u'fo\u00f5': [u'c\u00f5nt1']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the second set but not the first:
u'fo\xf5'

======================================================================
FAIL: test_prune_status_handles_unlink_errors (test.unit.test_container_crawler.TestContainerCrawler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tburke/code/container-crawler/.tox/py27-minver/lib/python2.7/site-packages/mock/mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1120, in test_prune_status_handles_unlink_errors
    {u'fo\u00f5': [u'c\u00f5nt1', u'c\u00f5nt2']})
  File "/home/tburke/code/container-crawler/test/unit/test_container_crawler.py", line 1039, in _prune_files_test_helper
    set(expected.keys()).union(extra_files))
AssertionError: Items in the second set but not the first:
u'fo\xf5'
```

Doing this seemed more expedient and less likely to fail in the future than fixing up every place we set up configs.

File: test/unit/test_container_crawler.py (hunk 2 above)
Comment (tipabu) [reply test/unit/test_container_crawler.py]: Fixed up to be more like what I'd done in `test_enumerator_worker`. Ditto for L297.

File: tox.ini (hunk 3 above)
Comment (tipabu) [reply tox.ini]: Yeah, I can drop this -- I'd just done it because that's my usual way of testing with a range of versions of things.

Still, I think there'd be some value in having tests run against our stated min version of Swift...
